pipeline/trusted-resources/pkg/config/store_test.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakek8s "k8s.io/client-go/kubernetes/fake"
+	"knative.dev/pkg/configmap/informer"
+	logtesting "knative.dev/pkg/logging/testing"
+	rtesting "knative.dev/pkg/reconciler/testing"
+	"knative.dev/pkg/system"
+)
+
+func TestNewConfigStore(t *testing.T) {
+	ctx, _ := rtesting.SetupFakeContext(t)
+
+	ns := system.Namespace()
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      trustedTaskConfig,
+			Namespace: ns,
+		},
+	}
+	fakekubeclient := fakek8s.NewSimpleClientset(cm)
+	cmw := informer.NewInformedWatcher(fakekubeclient, system.Namespace())
+
+	store := NewConfigStore(logtesting.TestLogger(t))
+	store.WatchConfigs(cmw)
+	if err := cmw.Start(ctx.Done()); err != nil {
+		t.Fatalf("Error starting configmap.Watcher %v", err)
+	}
+
+	// Check that with an empty configmap we get the default values.
+	if diff := cmp.Diff(store.Load(), defaultConfig()); diff != "" {
+		t.Errorf("unexpected data: %v", diff)
+	}
+
+	cm.Data = map[string]string{}
+	cm.Data[cosignKey] = "newPath"
+
+	if _, err := fakekubeclient.CoreV1().ConfigMaps(ns).Update(ctx, cm, metav1.UpdateOptions{}); err != nil {
+		t.Errorf("error updating configmap: %v", err)
+	}
+
+	// It should be updated by then...
+	time.Sleep(100 * time.Millisecond)
+
+	// Test that the values are set!
+	if diff := cmp.Diff("newPath", store.Load().Cosign.CosignKey); diff != "" {
+		t.Error(diff)
+	}
+
+	cfg := FromContext(store.ToContext(ctx))
+	if diff := cmp.Diff("newPath", cfg.Cosign.CosignKey); diff != "" {
+		t.Error(diff)
+	}
```

Comment:
There's not too much you gain by diffing strings - IIUC `cmp` only diffs on the field level. I'd change this to a diff against the fully defined struct for the config.

Reply:
sure, thanks!